pickle/SymbolTable.java

```diff
@@ -4,31 +4,56 @@
 
 public class SymbolTable {
 
-	private HashMap<String, HashMap<String, STEntry>> tables;		// HashMap to hold all symbol tables
-	private HashMap<String, STEntry> globalST;						// Global symbol table
+	private HashMap<String, STEntry> globalST;		// Global symbol table
+	private final int VAR_ARGS = -1;				// Value for variable args
 
-	public SymbolTable() {
-		this.tables = new HashMap<>();
+	/**
+	 *
+	 * Returns a SymbolTable object that holds the symbol tables of the pickle code
+	 * <p>
+	 *
+	 */
+	public SymbolTable()
+	{
 		this.globalST = new HashMap<>();
 		initGlobal();
-		this.tables.put("global", globalST);
 	}
 
-	public STEntry getSymbol(String symbol) {
-		//check correct symbol table(s?) for requested symbol
-
-		return null;
+	/**
+	 *
+	 * Returns the symbol table entry for a given symbol
+	 * <p>
+	 *
+	 * @param symbol	Name of symbol in the symbol table
+	 * @return			Symbol table entry (STEntry), null if not found
+	 */
+	public STEntry getSymbol(String symbol)
+	{
+		return this.globalST.get(symbol);
```

Comment:
the Map.get() method will return null on an entry not existing, wont this throw a runtime error trying to return null instead of STEntry?

Reply:
Pretty sure since it's and object the reference can be null

Reply:
have you tested?

Reply:
dont thumbs down me

Reply:
thanks

Reply:
would probably be much better as the following example, prevents passing around null references and ensures function always returns an STEntry object

`
return (this.globalST.containsKey(symbol) ) ? this.globalST.get(symbol) : new STEntry("", Classif.EMPTY);
`

Reply:
This will also help the Scanner because I'll want to check the primary classification of the STEntry object and I wont have to add extra code to first check if the object is null and then check the primary classification

Reply:
same thing as checking if classif is empty, but if that is how you want it, that is how you will get it

Reply:
I mean you're not wrong there still has to be a check to see if it exists in the symbolTable, but being able to check the primary classification allows consistency in other portions of the program, not just in the scanner